src/touchscreen_list_menu.c

```diff
@@ -0,0 +1,429 @@
+#include "touchscreen_list_menu.h"
+
+#include "global.h"
+
+#include "constants/sndseq.h"
+
+#include "data/sbox_gra.naix"
+
+#include "font.h"
+#include "gf_gfx_loader.h"
+#include "palette.h"
+#include "sys_task_api.h"
+#include "text.h"
+#include "unk_02005D10.h"
+
+static TouchscreenListMenu *TouchscreenListMenu_CreateInternal(TouchscreenListMenuSpawner *spawner, TouchscreenListMenuHeader *header, u8 isTouch, u8 x, u8 y, u8 width, u8 selection, TouchscreenListMenuCallback callback, void *callbackArg, BOOL silent, int alignment);
+static void TouchscreenListMenuSpawner_ScheduleLoadGraphicsToVram(TouchscreenListMenuSpawner *spawner, TouchscreenListMenuHeader *header, PaletteData *paletteData, HeapID heapId);
+static void Task_LoadTouchscreenListMenuGraphicsToVram(SysTask *task, void *taskData);
+static u8 TouchscreenListMenu_GetItemsTextMaxWidth(LISTMENUITEM *listMenuItem, u8 num, FontID fontId, u8 margin);
+static void TouchscreenListMenu_CreateWindows(TouchscreenListMenu *menu);
+static void TouchscreenListMenu_DeleteWindows(TouchscreenListMenu *menu);
+static void TouchscreenListMenu_DrawButtons(TouchscreenListMenu *menu);
+static void TouchscreenListMenu_PrintOptions(TouchscreenListMenu *menu);
+static void TouchscreenListMenu_ToggleButtonPalette(TouchscreenListMenu *menu, u8 index, BOOL selected);
+static void TouchscreenListMenu_EraseTilemap(TouchscreenListMenu *menu);
+static void TouchscreenListMenu_DrawButtonsAndTransfer(TouchscreenListMenu *menu, u8 cursorPos);
+static int TouchscreenListMenu_HandleTouchInput(TouchscreenListMenu *menu, BOOL *flagRet);
+static int TouchscreenListMenu_HandleKeyInput(TouchscreenListMenu *menu);
+static void TouchscreenListMenu_PlaySE(TouchscreenListMenu *menu, u16 sndseq);
+static void TouchscreenListMenu_InvokeCallback(TouchscreenListMenu *menu, int a1);
+
+TouchscreenListMenuSpawner *TouchscreenListMenuSpawner_Create(HeapID heapId, PaletteData *paletteData) {
```

Comment:
Ideally just rename spawner to manager and remove the current manager as a struct